### PR TITLE
feat: Add disconnect method to event hub

### PR DESCRIPTION
**Pull request by gismya**
- [x] I have added automatic tests where applicable
- [x] The PR title is suitable as a release note
- [x] The PR contains a description of what has been changed
- [ ] The description contains manual test instructions

## Changes

PR to address #238. Adds a disconnect method to the event hub

### CODE REVIEW

**Comment by ffMathy** [comment]
Thank you so much for this! :heart:

**Review by ffMathy — APPROVED** [review]

File: test/event_hub.test.ts [hunk 1]
```diff
@@ -239,6 +241,11 @@ describe("EventHub", () => {
     };
     expect(EventData).toEqual(expectedEvent);
   });
+
+  test("Disconnecting should disconnect the socket", () => {
+    eventHub.disconnect();
+    expect(disconnectCalled).toBe(true);
```
Comment (jimmycallin) [inline test/event_hub.test.ts:247]:
```suggestion
    expect(eventHub._socketIo.discunnect).toBeCalled();
```

think you can do something like this, a bit more idiomatic

Comment (gismya) [reply test/event_hub.test.ts:247]: No, I'm setting `_socketIo` to undefined as part of the disconnect.

Comment (jimmycallin) [reply test/event_hub.test.ts:247]: ah, I guess you then could do this:

```
const disconnectFn = vi.fn()
...
disconnect: disconnectFn,

expect(disconnectFn).toHaveBeenCalled()
```

but we can go with this approach as well

**Review by jimmycallin — APPROVED** [review]

**Pull request merged**